client/modules/IDE/hooks/useSketchActions.js

```diff
@@ -41,7 +41,9 @@ const useSketchActions = () => {
 
   function downloadSketch() {
     dispatch(autosaveProject());
-    exportProjectAsZip(project.id);
+    if (authenticated && project.owner === authenticated.userId) {
+      exportProjectAsZip(project.id);
+    }
```

Comment:
I think you've got the `if` checking a little bit backwards here.  We want to allow downloading of other people's projects.  But we do not want to call save functions on other people's projects.

Should be more like
```
 if (authenticated && project.owner === authenticated.userId) {
  dispatch(autosaveProject());
 }   
 exportProjectAsZip(project.id);
```

Reply:
@lindapaiste ,I've revised the downloadSketch function to ensure that we only trigger the autosave functionality if the user is authenticated and they are the owner of the sketch. Here's the updated implementation:
 
 function downloadSketch() {
    if (authenticated && project.owner === authenticated.userId) {
        dispatch(autosaveProject());
    }
    exportProjectAsZip(project.id);
}

Reply:
@lindapaiste , And I think could also be done , correct me if I'm wrong

if (authenticated && project.owner === authenticated.userId) {
  dispatch(autosaveProject());
  exportProjectAsZip(project.id);
} else {
  exportProjectAsZip(project.id);
}

This code block first checks if the user is authenticated and if the current user is the owner of the project. If both conditions are met, it triggers the autosaveProject function before exporting the project as a zip file. If the conditions are not met, it directly exports the project as a zip file without triggering the autosave function.

Reply:
You should run your code locally and make sure that it works before submitting a PR.  `authenticated.userId` will not work because `authenticated` is a `boolean` value -- it is not the user object.  `project.owner` is not the id -- it's the entire user object for the owner.  We need a different conditional check here.  Probably `authenticated && canEditProjectName` would do it?  But we need to try it out and check the various situations, like if this is a new sketch that has not been saved before.